sympy/plotting/plot.py

```diff
@@ -1044,47 +1168,64 @@ def process_series(self):
                 raise ValueError(
                 "All numbers from ylim={} must be finite".format(ylim))
             ylim = (float(i) for i in ylim)
-            self.ax.set_ylim(ylim)
-        if not isinstance(self.ax, Axes3D) or self.matplotlib.__version__ >= '1.2.0':  # XXX in the distant future remove this check
-            self.ax.set_autoscale_on(parent.autoscale)
+            ax.set_ylim(ylim)
+        if not isinstance(ax, Axes3D) or self.matplotlib.__version__ >= '1.2.0':  # XXX in the distant future remove this check
+            ax.set_autoscale_on(parent.autoscale)
         if parent.axis_center:
             val = parent.axis_center
-            if isinstance(self.ax, Axes3D):
+            if isinstance(ax, Axes3D):
                 pass
             elif val == 'center':
-                self.ax.spines['left'].set_position('center')
-                self.ax.spines['bottom'].set_position('center')
+                ax.spines['left'].set_position('center')
+                ax.spines['bottom'].set_position('center')
             elif val == 'auto':
-                xl, xh = self.ax.get_xlim()
-                yl, yh = self.ax.get_ylim()
+                xl, xh = ax.get_xlim()
+                yl, yh = ax.get_ylim()
                 pos_left = ('data', 0) if xl*xh <= 0 else 'center'
                 pos_bottom = ('data', 0) if yl*yh <= 0 else 'center'
-                self.ax.spines['left'].set_position(pos_left)
-                self.ax.spines['bottom'].set_position(pos_bottom)
+                ax.spines['left'].set_position(pos_left)
+                ax.spines['bottom'].set_position(pos_bottom)
             else:
-                self.ax.spines['left'].set_position(('data', val[0]))
-                self.ax.spines['bottom'].set_position(('data', val[1]))
+                ax.spines['left'].set_position(('data', val[0]))
+                ax.spines['bottom'].set_position(('data', val[1]))
         if not parent.axis:
-            self.ax.set_axis_off()
+            ax.set_axis_off()
         if parent.legend:
-            if self.ax.legend():
-                self.ax.legend_.set_visible(parent.legend)
+            if ax.legend():
+                ax.legend_.set_visible(parent.legend)
         if parent.margin:
-            self.ax.set_xmargin(parent.margin)
-            self.ax.set_ymargin(parent.margin)
+            ax.set_xmargin(parent.margin)
+            ax.set_ymargin(parent.margin)
         if parent.title:
-            self.ax.set_title(parent.title)
+            ax.set_title(parent.title)
         if parent.xlabel:
-            self.ax.set_xlabel(parent.xlabel, position=(1, 0))
+            ax.set_xlabel(parent.xlabel, position=(1, 0))
         if parent.ylabel:
-            self.ax.set_ylabel(parent.ylabel, position=(0, 1))
+            ax.set_ylabel(parent.ylabel, position=(0, 1))
+
+    def process_series(self):
```

Comment:
Does this need to be public? Why would a user need to call this? If it needs to be public, it needs a docstring.

Reply:
The user is not really expected to touch this class as a whole at all (it is a class implementing the backend "protocol"). The class itself is not exported in the `__init__.py` file. I think it is fair to leave the naming as it is (an additional underscore seems unnecessary if the user never sees the class anyway; developers that will be implementing new backends probably also would prefer no underscore so they can immediately see that they are expected to provide this method).

Reply:
Should we make the class private then?

Reply:
Also, how will the user create these subplots then? I don't see any other function or modifications to plot() that allow user interaction.

Reply:
I am uncertain what a private class means in python (it is not exported in the sympy namespace and I thought that was a typical way to mark something as "somewhat private").

As for user interaction: the user creates a non-grid plots with `plot*` or `Plot`, the user can change some aesthetics by accessing various properties of the newly created `Plot` instance, and can call `show` or `save` (all of this is part of the `Plot` class). The `Plot` class has an internal (`._backend`) property referring to the instance of the class implementing the drawing backend (usually `matplotlib`). The user is not expected to know about `_backend` unless they want to act on the matplotlib object itself (which is presumably none of sympy's duties/business). For a grid plot the user starts the same way, but then assembles the independent plots in a `PlotGrid` object.

Reply:
I only mean "private" in the sense that it has a preprended underscore and we don't include it in the documented API, thus it can be changed, deleted, etc without deprecation.

> For a grid plot the user starts the same way, but then assembles the independent plots in a PlotGrid object.

That's what I thought. The user does have to actually type `PlotGrid`. I was confused when you said the user doesn't touch this class above. I would say they do touch this class. So it should not be private. I do still recommend `process_series` have a prepended underscore though.

Reply:
Isn't `process_series` a method of `MatplotlibBackend`, not `PlotGrid`? `MatplotlibBackend` and the other backends are not public classes. Does this address your worries?

Reply:
Maybe it is. Github squishes everythign and it isn't always clear what portion of code I'm looking at. Sorry for the mistake. Long conversation based on a non-issue. Go forth ! :)